syntax = "proto3";

package auth;

service AuthService {
    rpc Login (LoginRequest) returns (LoginReply);
    rpc Register (RegisterRequest) returns (RegisterReply);
    rpc Refresh (RefreshRequest) returns (RefreshReply);
    // rpc ValidateToken();
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginReply {
    string access_token = 1;
    string refresh_token = 2;
}

message RegisterRequest {
    string username = 1;
    string password = 2;
}

message RegisterReply {
    bool success = 1;
    string message = 2; 
}

message RefreshRequest {
    string refresh_token = 1;
}

message RefreshReply {
    string access_token = 1;
}


// message ValidateTokenRequest{
//     string token = 1;
// };

// message ValidateTokenResponse{
//     string token = 1;
// };

